homeassistant/components/schedule/__init__.py

```diff
@@ -109,16 +110,22 @@ def serialize_to_time(value: Any) -> Any:
     vol.Optional(CONF_ICON): cv.icon,
 }
 
+# Extra data that the user can set on each time range
+CUSTOM_DATA_VALUE = vol.Or(bool, str, int, float)
+CUSTOM_DATA_SCHEMA = vol.Or(vol.Schema({str: CUSTOM_DATA_VALUE}), CUSTOM_DATA_VALUE)
```

Comment:
Let's use `vol.Any`, that's almost exclusively used throughout the codebase
```suggestion
CUSTOM_DATA_VALUE = vol.Any(bool, str, int, float)
CUSTOM_DATA_SCHEMA = vol.Any(vol.Schema({str: CUSTOM_DATA_VALUE}), CUSTOM_DATA_VALUE)
```